bin/nvcc_wrapper

```diff
@@ -194,28 +200,48 @@ do
   -rdc=*|-maxrregcount*|--maxrregcount*)
     cuda_args="$cuda_args $1"
     ;;
-  #Handle unsupported standard flags
-  --std=c++1y|-std=c++1y|--std=c++1z|-std=c++1z|--std=gnu++1y|-std=gnu++1y|--std=gnu++1z|-std=gnu++1z|--std=c++2a|-std=c++2a|--std=c++17|-std=c++17)
-    fallback_std_flag="-std=c++14"
-    # this is hopefully just occurring in a downstream project during CMake feature tests
-    # we really have no choice here but to accept the flag and change  to an accepted C++ standard
-    echo "nvcc_wrapper does not accept standard flags $1 since partial standard flags and standards after C++14 are not supported. nvcc_wrapper will use $fallback_std_flag instead. It is undefined behavior to use this flag. This should only be occurring during CMake configuration."
-    if [ -n "$std_flag" ]; then
-       warn_std_flag
-       shared_args=${shared_args/ $std_flag/}
+  #Handle c++11
+  --std=c++1y|-std=c++1y|--std=c++1z|-std=c++1z|--std=gnu++1y|-std=gnu++1y|--std=gnu++1z|-std=gnu++1z|--std=c++17|-std=c++17|--std=c++2a|-std=c++2a|--std=gnu++17|-std=gnu++17)
+    if [ $host_only -eq 1 ]; then
+      #accept the flag as is for host compiler
+      if [ -n "$std_flag" ]; then
+         warn_std_flag
+         shared_args=${shared_args/ $std_flag/}
+      fi
+      std_flag=$1
+      shared_args="$shared_args $std_flag"
+    else
+      #nope, not okay, we are in nvcc mode
+      bad_std_flag=$1
+      bad_nvcc_std=1
+      if [ -n "$std_flag" ]; then
+         warn_std_flag
+         shared_args=${shared_args/ $std_flag/}
+      fi
+      std_flag=$1
```

Comment:
Doesn't this change the behavior where we earlier used fallback_std_flag?

Reply:
Correct. Same as comment below. CMake tests now run in host mode so we don't need to do any fallback flag hacking.